pennylane/gradients/vjp.py

```diff
@@ -153,11 +158,14 @@ def vjp(tape, dy, gradient_fn, gradient_kwargs=None):
         # is simply none.
         return [], lambda _: None
 
-    if math.allclose(dy, 0):
-        # If the dy vector is zero, then the
-        # corresponding element of the VJP will be zero,
-        # and we can avoid a quantum computation.
-        return [], lambda _: math.convert_like(np.zeros([num_params]), dy)
+    try:
+        if math.allclose(dy, 0):
+            # If the dy vector is zero, then the
+            # corresponding element of the VJP will be zero,
+            # and we can avoid a quantum computation.
+            return [], lambda _: math.convert_like(np.zeros([num_params]), dy)
+    except AttributeError:

```

Comment:
Oh, so this is really annoying. Newer versions of TF will attempt to vectorize Jacobian computations by default, and as part of this vectorization process, they trace the cost function. The issue is:

- you can't vectorize a quantum execution, even though TF tries 😆 So it's a very pointless step that does nothing but adds overhead.

- During tracing, it sends proxy variables that have no value; e.g., you can't call `dy.numpy()` (since the value doesn't exist yet).

The second bullet point is the cause of the attribute error --- `math.allclose` is calling `dy.numpy()`, which doesn't exist in vectorized mode.

I attempted to rewrite `math.allclose()` to directly implement `tf.abs(a-b) <= atol + b * rtol` as per the definition, but then ran into another error - TF was complaining that a proxy variable cannot be used in a Python conditional 🙁

Reply:
Basically: we need the vectorization to 'work' from TF's perspective, even though it has no effect.